src/model/trace-viewer.js

```diff
@@ -13,7 +13,7 @@
 // limitations under the License.
 
 // eslint-disable-next-line import/prefer-default-export
-export function getTraceName(spans, processes) {
-  const span = spans.find(sp => sp.spanID === sp.traceID) || spans[0];
-  return span ? `${processes[span.processID].serviceName}: ${span.operationName}` : '';
+export function getTraceName(spans) {
+  const span = spans.filter(sp => !sp.references || !sp.references.length)[0];
+  return span ? `${span.process.serviceName}: ${span.operationName}` : '';
```

Comment:
should this also fallback to FALLBACK_TRACE_NAME?

Reply:
It can, but that is deferred, the fallback is used closer to the view — the use of the fallback or not is a presentation decision.